csharp-basic-topics/RemoveIllegalCharsFromPathAndFilenames/RemoveIllegalCharsFromPathAndFilenames/StringMethods.cs

```diff
@@ -0,0 +1,62 @@
+﻿public static class StringMethods
+{
+    public static List<string> CheckForInvalid(List<string> strings, HashSet<char> criteria)
+    {
+        List<string> result = new List<string>();
+
+        foreach (var str in strings)
```

Comment:
Isn't this

`return strings.Where(str => criteria.Any(ch => str.Contains(ch));`

You can change `List<string>` return type to `IEnumerable<string>`